lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +157,69 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+*/
+
+function getSolutionsAndSymbol (equation) {
+  const left = equation.leftNode;
+  const rightNode = Node.Creator.constant(0);
+
+  const solutions = [];
+
+  let symbol, steps, factorsWithSymbols;
+
+  // If left hand side is a power node and it does not resolve to a constant (e.g. 2^7)
+  // then it is a factor. If left hand side is a multiplication node, return a list of all the valid factors.
```

Comment:
small thing, but can you make these lines max 80 chars?

and `(e.g. 2^7 resolves to a constant so it is not a factor, but x^2 would have factors x=0)` would be more clear, as I was reading this I thought you meant 2^7 is a power node that doesn't resolve to a constant which confused me